addons/api/mirage/factories/host-catalog.js

```diff
@@ -11,25 +11,27 @@ const pluginTypes = ['aws', 'azure', 'foobar'];
 let pluginTypeCounter = 1;
 
 export default factory.extend({
+  id: (i) => `host-catalog-id-${i}`,
+
+  // Cycle through available types
+  type: (i) => types[i % types.length],
+
   authorized_actions: () =>
     permissions.authorizedActionsFor('host-catalog') || [
       'no-op',
       'read',
       'update',
       'delete',
     ],
-  authorized_collection_actions: () => {
+  authorized_collection_actions: function () {
+    const isStatic = this.type === 'static';
```

Comment:
I noticed that this function changed from an anonymous function to a named one. Is this simply so it gains access to `this`? That is the only thing different that I can see as to why it would need to change.

Reply:
Yes exactly, because this field is dependent on `type`, whereas previously it was not.